libraries/protocol/include/steem/protocol/types.hpp

```diff
@@ -148,6 +148,11 @@ namespace steem {
          friend bool operator == ( const extended_private_key_type& p1, const extended_private_key_type& p2);
          friend bool operator != ( const extended_private_key_type& p1, const extended_private_key_type& p2);
       };
+
+      extern std::string chain_id_name;
```

Comment:
Can this not be a global variable?

I think everywhere that needs the chain id probably has access to the database. Make it a settable parameter in the database. When the chain_plugin opens the database, set it to STEEM_CHAIN_ID if it is not set via CLI, or the passed arg.

The one place that this might not work well is the p2p code. If needed, add a `get_chain_id()` virtual method to the node delegate and implement it in the p2p plugin.

Reply:
Question 1:
Where should I get info about `STEEM_CHAIN_ID`, where I use wallet ? It is necessarily in `sign_transaction` function in `wallet.cpp`.
My idea is, to add additional method to `condenser_api` for example: `get_chain_id_name` and calling `_remote_api->get_chain_id_name()`, when we are first time in `sign_transaction`.

Question 2:
In file `sign_transaction.cpp` in `main` function we havent't access to `database` object. I think, I should create `fc::sha256` object independently from `database` object.

Reply:
1. Add a command line arg to the wallet as well.

2. Keep the default STEEM_CHAIN_ID.

One purpose of the chain ID is to segment transactions between chains via signature. It would be bad if the cli wallet automatically worked if you connected to GOLOS. Right now you need to recompile for it to work. This is to add the ability for the user to configure the same binary to work on compatible forks. The sign_transaction utility is not widely used to the best of my knowledge, so I think it is safe keeping it working with just the default chain id. This is only going to be used for test nets, not for the main net. The chain id for the main net is set and will never change.